lib/auth/winwebauthn/webauthn_windows.go

```diff
@@ -0,0 +1,945 @@
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//go:build windows
+// +build windows
+
+package winwebauthn
+
+import (
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	"syscall"
+	"unsafe"
+
+	"github.com/duo-labs/webauthn/protocol"
+	wanlib "github.com/gravitational/teleport/lib/auth/webauthn"
+	log "github.com/sirupsen/logrus"
+	"golang.org/x/sys/windows"
+)
+
+var (
+	modWebAuthn = windows.NewLazySystemDLL("WebAuthn.dll")
+
+	procWebAuthNGetApiVersionNumber                           = modWebAuthn.NewProc("WebAuthNGetApiVersionNumber")
+	procWebAuthNIsUserVerifyingPlatformAuthenticatorAvailable = modWebAuthn.NewProc("WebAuthNIsUserVerifyingPlatformAuthenticatorAvailable")
+	procWebAuthNAuthenticatorMakeCredential                   = modWebAuthn.NewProc("WebAuthNAuthenticatorMakeCredential")
+	procWebAuthNFreeCredentialAttestation                     = modWebAuthn.NewProc("WebAuthNFreeCredentialAttestation")
+	procWebAuthNAuthenticatorGetAssertion                     = modWebAuthn.NewProc("WebAuthNAuthenticatorGetAssertion")
+	procWebAuthNFreeAssertion                                 = modWebAuthn.NewProc("WebAuthNFreeAssertion")
+	procWebAuthNGetErrorName                                  = modWebAuthn.NewProc("WebAuthNGetErrorName")
+
+	moduser32               = windows.NewLazySystemDLL("user32.dll")
+	procGetForegroundWindow = moduser32.NewProc("GetForegroundWindow")
+)
+
+type client struct {
+	version           int
+	hasCompileSupport bool
+	isAvailable       bool
+	hasPlatformUV     bool
+}
+
+// initClient creates client which contains diagnostics info.
+// Diagnostics are safe to cache because dll isn't something that
+// could change during program invocation.
+// Client will be alaways created, even if dll is missing on system.
+func initClient() *client {
+	v, err := checkIfDLLExistsAndGetAPIVersionNumber()
+	if err != nil {
+		log.WithError(err).Warn("Windows webauthn failed to check version")
+		return &client{
+			hasCompileSupport: true,
+			isAvailable:       false,
+		}
+	}
+	uvPlatform, err := isUVPlatformAuthenticatorAvailable()
+	if err != nil {
+		// This should not happen if dll exists, however we are fine with
+		// to proceed without uvPlatform.
+		log.WithError(err).Warn("Windows webauthn failed to check isUVPlatformAuthenticatorAvailable")
+	}
+
+	return &client{
+		version:           v,
+		hasCompileSupport: true,
+		hasPlatformUV:     uvPlatform,
+		isAvailable:       v > 0 && v <= 4,
```

Comment:
I have done that on purpose (but forgot to add comment for it). I think it will be better to test it when v5 goes out instead of blindly supporting it. What's your opinion on that?

Reply:
Great, thanks.